packages/postcss-reduce-initial/src/script/__tests__/io.js

```diff
@@ -0,0 +1,86 @@
+import nock from 'nock';
+import { handleError, toJSONString, write, generate } from '../lib/io.mjs';
+import testData from './sampleProperties.json';
+
+describe('toJSONString', () => {
+  const rawData = {
+    foo: 'bar',
+    'baz-qux': 'quux',
+    quuz: 'corge',
+    garply: 'waldo',
+    'fred-plugh': 'xyzzy-thud',
+  };
+
+  test('should have correct data type (string)', () => {
+    expect(typeof toJSONString(rawData)).toBe('string');
```

Comment:
I think my logic was that the intended output of the function `toJSONString` is -- in the crudest sense from it's name -- a string. So if someone were to refactor the code later and lets say... omit the `JSON.stringify` and return a `Buffer` like you mention... it would diverge from what the function is supposed to do.

That return could still be parsable by `JSON.parse` and pass the next test - but theoretically might not be compatible with whatever is responsible for saving the file (which might not be `fs.writeFile`) if that expects a string. Or some other part of the code that in some future revision relies on the output from `toJSONString` to be a string.

```js
const data = { foo: 'bar', baz: 'qux' };
const returnValue = Buffer.from(JSON.stringify(data));
console.log(JSON.parse(returnValue)); // { foo: 'bar', baz: 'qux' }
console.log(returnValue.trim()); // TypeError
```

It's possible I'm erring a bit too much on the side of caution though. What do you think?

Reply:
I think this test is not that necessary, as I do not see the code to write the file changing that often. Suppose someone would change that code anyway, why would they assume they are receiving a string?

Reply:
Roger that - I'll drop it then 👍